source/Lib/Utilities/NoMallocThreadPool.cpp

```diff
@@ -270,6 +274,77 @@ bool NoMallocThreadPool::processTask( int threadId, NoMallocThreadPool::Slot& ta
   return true;
 }
 
+#ifdef HAVE_PTHREADS
+
+template<class TFunc, class... TArgs>
+NoMallocThreadPool::PThread::PThread( TFunc&& func, TArgs&&... args )
+{
+  using WrappedCall     = std::function<void()>;
+  std::unique_ptr<WrappedCall> call = std::make_unique<WrappedCall>( std::bind( func, args... ) );
+
+  using PThreadsStartFn = void* (*) ( void* );
+  PThreadsStartFn threadFn = []( void* p ) -> void*
+  {
+    std::unique_ptr<WrappedCall> call( static_cast<WrappedCall*>( p ) );
+
+    ( *call )();
+
+    return nullptr;
+  };
+
+  pthread_attr_t attr;
+  CHECK( pthread_attr_init( &attr ) != 0, "pthread_attr_init() failed" );
```

Comment:
please don't put actual functionality into CHECK()-Macro. Those might be stripped down eventually. Only perform checks there, don't put actual code there.